components/brave_vpn/brave_vpn_service.cc

```diff
@@ -1015,6 +1016,12 @@ void BraveVpnService::OnPrepareCredentialsPresentation(
     FetchRegionData(false);
   }
 
+  if (!IsNetworkAvailable()) {
```

Comment:
Curious why connect failed setting is needed here?
Do you want to set it If we have network failure during the loading purchased state?

Reply:
and do you think purchased state also should be changed to (failed?) when network failed?

Reply:
We have a call form the panel to get credentials. Since the credentials are stored in prefs it goes here and trying to connect which takes some time and user sees loading throbber for some time. Probably better idea to put it into `GetBraveVPNConnectionAPI()->CheckConnection` wdyt?

Reply:
moved it to `LoadPurchasedState` as discussed in Slack